File: ios/Stopwatch/core/use_cases/impl/UpdateStopwatchTimeUseCaseImpl.swift
class UpdateStopwatchTimeUseCaseImpl<MSS>: UpdateStopwatchTimeUseCase
where MSS: MutableStateStore, MSS.State == StopwatchState {
    private let store: MSS
    
    init(_ store: MSS) {
        self.store = store
    }
    
    func execute(_ timerState: TimerState) async {
        await store.update {
            StopwatchState(
                status: $0.status,
                milliseconds: timerState.milliseconds,
                completedLaps: $0.completedLaps,
                completedLapsMilliseconds: $0.completedLapsMilliseconds
            )
        }
    }
}

File: android/app/src/main/java/com/rafaeltmbr/stopwatch/core/use_cases/impl/UpdateStopwatchTimeUseCaseImpl.kt
package com.rafaeltmbr.stopwatch.core.use_cases.impl

import com.rafaeltmbr.stopwatch.core.data.stores.MutableStateStore
import com.rafaeltmbr.stopwatch.core.entities.StopwatchState
import com.rafaeltmbr.stopwatch.core.entities.Time
import com.rafaeltmbr.stopwatch.core.services.TimerService
import com.rafaeltmbr.stopwatch.core.use_cases.UpdateStopwatchTimeUseCase

class UpdateStopwatchTimeUseCaseImpl(
    private val store: MutableStateStore<StopwatchState>,
) : UpdateStopwatchTimeUseCase {
    override suspend fun execute(timerState: TimerService.State) {
        store.update {
            it.copy().updateTime(Time(timerState.milliseconds))
        }
    }
}